cpp/build-support/trim-boost.sh

```diff
@@ -22,7 +22,15 @@
 # so that we don't have to download the whole big boost project when we build
 # boost from source.
 #
-# After running this script, run upload-boost.sh to put the bundle on bintray
+# To test building Arrow locally with the boost bundle this creates, add:
+#
+#     set(BOOST_SOURCE_URL /path/to/arrow/cpp/build-support/boost_1_75_0/boost_1_75_0.tar.gz)
+#
+# to the beginning of the build_boost() macro in ThirdpartyToolchain.cmake
```

Comment:
`ARROW_BOOST_URL=/path/to/arrow/cpp/build-support/boost_1_75_0/boost_1_75_0.tar.gz cmake ...` will work.